Bottom: 4c0c5940bb7a52d83645c42a500c3854626a67e1
Top:    e57caf8778ecad03e0c03a181feb8d3a16e8e014
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-30 15:20:40 +0100

Remove live drawing parameters for HID->use_mask()

These are no longer used by the autorouter, so lets clean the use_mask()
interface to JUST deal with _masking_ from now on ;)


---

diff --git a/src/hid.h b/src/hid.h
index 820760e..9104190 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -341,12 +341,6 @@ typedef enum
 #define HID_MASK_CLEAR 2
     /* Polygons being drawn after clears.  */
 #define HID_MASK_AFTER 3
-    /* Set to do live drawing on the screen   */
-#define HID_LIVE_DRAWING 4
-    /* stop live drawing on the screen   */
-#define HID_LIVE_DRAWING_OFF 5
-    /* flush any queued drawing   */
-#define HID_FLUSH_DRAW_Q 6
 
     /* Set a color.  Names can be like "red" or "#rrggbb" or special
        names like "erase".  *Always* use the "erase" color for removing
diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 45f1e96..a28b19a 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -259,23 +259,6 @@ ghid_use_mask (int use_it)
   GdkColor color;
   render_priv *priv = gport->render_priv;
 
-  if (use_it == HID_FLUSH_DRAW_Q)
-    {
-      gdk_flush ();
-      return;
-    }
-  else if (use_it == HID_LIVE_DRAWING)
-    {
-      old = gport->drawable;
-      gport->drawable = gport->drawing_area->window;
-      return;
-    }
-  else if (use_it == HID_LIVE_DRAWING_OFF)
-    {
-      gport->drawable = old;
-      return;
-    }
-
   if (!gport->pixmap)
     return;
   if (use_it == cur_mask)
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 3922fc4..776a345 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -3041,22 +3041,6 @@ lesstif_use_mask (int use_it)
 {
   static Window old;
 
-  if (use_it == HID_FLUSH_DRAW_Q)
-    {
-      XFlush (display);
-      return;
-    }
-  else if (use_it == HID_LIVE_DRAWING)
-    {
-      old = pixmap;
-      pixmap = window;
-      return;
-    }
-  else if (use_it == HID_LIVE_DRAWING_OFF)
-    {
-      pixmap = old;
-      return;
-    }
   if ((TEST_FLAG (THINDRAWFLAG, PCB) || TEST_FLAG(THINDRAWPOLYFLAG, PCB)) &&
       !use_xrender)
     use_it = 0;
